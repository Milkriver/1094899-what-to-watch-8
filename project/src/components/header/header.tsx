import { connect, ConnectedProps } from 'react-redux';
import { Link } from 'react-router-dom';
import { IState } from '../../types/state';

const mapStateToProps = ({ authorizationStatus }: IState) => ({
  authorizationStatus,
});
const connector = connect(mapStateToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;
function Header(props: PropsFromRedux): JSX.Element {
  const { authorizationStatus } = props;
  // eslint-disable-next-line no-console
  console.log(authorizationStatus);
  return (
    <header className="page-header film-card__head">
      <div className="logo">
        <Link to={'/'} className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>
      {
        (authorizationStatus === 'AUTH') ?

          < ul className="user-block">
            <li className="user-block__item">
              <div className="user-block__avatar">
                <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
              </div>
            </li>
            <li className="user-block__item">
              <div>Mail</div>
              <a href="/#" className="user-block__link">Sign out</a>
            </li>
          </ul>
          :
          <div className="user-block">
            <a href="sign-in.html" className="user-block__link">Sign in</a>
          </div>
      }
    </header >
  );
}

export { Header };
export default connector(Header);
