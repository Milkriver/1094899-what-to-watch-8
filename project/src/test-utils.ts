import thunk, { ThunkDispatch } from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import { createAPI } from './services/api';
import { configureMockStore, MockStoreCreator } from '@jedmao/redux-mock-store';
import { IState } from './types/state';
import { Action } from 'redux';
import { AxiosInstance } from 'axios';

interface IFakeInfrastructure {
  mockAPI: MockAdapter
  mockStore: MockStoreCreator<
    IState,
    Action,
    ThunkDispatch<IState, AxiosInstance, Action>
  >
}

export const makeFakeInfrastructure = (): IFakeInfrastructure => {
  const onFakeUnauthorized = jest.fn();
  const api = createAPI(onFakeUnauthorized());
  const mockAPI = new MockAdapter(api);
  const middlewares = [thunk.withExtraArgument(api)];

  const mockStore: MockStoreCreator<
    IState,
    Action,
    ThunkDispatch<IState, AxiosInstance, Action>
  > = configureMockStore(middlewares);

  mockAPI
    .onAny()
    .reply(200, {});

  return {
    mockAPI,
    mockStore,
  };
};
