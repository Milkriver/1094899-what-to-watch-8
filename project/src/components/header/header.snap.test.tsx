import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { makeFakeInfrastructure } from '../../test-utils';
import { Header } from './header';


describe('Component: Header', () => {
  const { mockStore } = makeFakeInfrastructure();

  const store = mockStore({
    MOVIES: {
      movie: {
        id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '',
        backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''],
        runTime: 1, genre: '', released: 1, isFavorite: true,
      },
    },
    USER: {},
    REVIEWS: {},
  });

  test('should render correctly', () => {
    const history = createMemoryHistory();
    const page = (
      <Provider store={store}>
        <Router history={history}>
          <Header />;
        </Router>
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
