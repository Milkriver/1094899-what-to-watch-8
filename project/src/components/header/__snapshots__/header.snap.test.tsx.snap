// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render correctly 1`] = `
<div>
  <header
    class="page-header film-card__head"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="user-block"
    >
      <a
        class="user-block__link"
        href="/login"
      >
        Sign in
      </a>
    </div>
  </header>
  ;
</div>
`;
